controllers/vault_reconciler.go

```diff
@@ -181,261 +197,199 @@ func (r *WorkshopReconciler) addVaultAgentInjector(workshop *workshopv1.Workshop
-		if err := r.Delete(context.TODO(), internalService); err != nil {
-			return reconcile.Result{}, err
-		}
-		log.Infof("Deleted %s internal Service", internalService.Name)
+	service := kubernetes.NewService(workshop, r.Scheme, VAULTSERVICENAME, VAULTNAMESPACENAME, VaultServerlabels, []string{"http", "internal"}, []int32{8200, 8201})
+	// Delete Service
+	if err := r.Delete(context.TODO(), service); err != nil {
+		return reconcile.Result{}, err
 	}
+	log.Infof("Deleted %s VaultServer Service", service.Name)
 
-	clusterRoleBinding := kubernetes.NewClusterRoleBindingSA(workshop, r.Scheme, "vault-server-binding", vaultNamespace.Name,
-		labels, serviceAccount.Name, "system:auth-delegator", "ClusterRole")
-	clusterRoleBindingFound := &rbac.ClusterRoleBinding{}
-	clusterRoleBindingErr := r.Get(context.TODO(), types.NamespacedName{Name: clusterRoleBinding.Name, Namespace: vaultNamespace.Name}, clusterRoleBindingFound)
-	if clusterRoleBindingErr == nil {
-		// Delete ClusterRole Binding
-		if err := r.Delete(context.TODO(), clusterRoleBinding); err != nil {
-			return reconcile.Result{}, err
-		}
-		log.Infof("Deleted %s ClusterRole Binding", clusterRoleBinding.Name)
+	internalService := kubernetes.NewService(workshop, r.Scheme, VAULTINTERNALSERVICENAME, VAULTNAMESPACENAME, VaultServerlabels, []string{"http", "internal"}, []int32{8200, 8201})
+	// Delete internal Service
+	if err := r.Delete(context.TODO(), internalService); err != nil {
+		return reconcile.Result{}, err
 	}
+	log.Infof("Deleted %s VaultServer internal Service", internalService.Name)
 
-	serviceAccountFound := &corev1.ServiceAccount{}
-	serviceAccountErr := r.Get(context.TODO(), types.NamespacedName{Name: serviceAccount.Name, Namespace: vaultNamespace.Name}, serviceAccountFound)
-	if serviceAccountErr == nil {
-		// Delete Service Account
-		if err := r.Delete(context.TODO(), serviceAccount); err != nil {
-			return reconcile.Result{}, err
-		}
-		log.Infof("Deleted %s Service Account", serviceAccount.Name)
+	clusterRoleBinding := kubernetes.NewClusterRoleBindingSA(workshop, r.Scheme, VAULTROLEBINDINGNAME, VAULTNAMESPACENAME,
+		VaultServerlabels, serviceAccount.Name, VAULTROLEBINDINGROLENAME, VAULTROLEBINDINGKINDNAME)
+	// Delete ClusterRole Binding
+	if err := r.Delete(context.TODO(), clusterRoleBinding); err != nil {
+		return reconcile.Result{}, err
 	}
+	log.Infof("Deleted %s  VaultServer ClusterRole Binding", clusterRoleBinding.Name)
 
-	configMap := kubernetes.NewConfigMap(workshop, r.Scheme, "vault-config", vaultNamespace.Name, labels, extraconfigFromValues)
-	configMapFound := &corev1.ConfigMap{}
-	configMapErr := r.Get(context.TODO(), types.NamespacedName{Name: configMap.Name, Namespace: vaultNamespace.Name}, configMapFound)
-	if configMapErr == nil {
-		// Delete configMap
-		if err := r.Delete(context.TODO(), configMap); err != nil {
-			return reconcile.Result{}, err
-		}
-		log.Infof("Deleted %s configMap", configMap.Name)
+	// Delete Service Account
```

Comment:
@MaazMS we also need to remove the user from privileged SCC; as it was added here: https://github.com/stakater/workshop-operator/blob/4175f97d81e39656518a9b4c2e36a3a653cf8527/controllers/vault_reconciler.go#L114-L127

so, for proper cleanup we need to remove it from everywhere